src/Components/Web/src/Forms/InputBase.cs

```diff
@@ -18,7 +18,7 @@ public abstract class InputBase<TValue> : ComponentBase, IDisposable
         private readonly EventHandler<ValidationStateChangedEventArgs> _validationStateChangedHandler;
         private bool _previousParsingAttemptFailed;
         private ValidationMessageStore _parsingValidationMessages;
-        private Type _nullableUnderlyingType;
+        protected Type _nullableUnderlyingType;
```

Comment:
This is a change to the public API surface and would need to go through an API review. This would mean changing this a property and documenting it, and I still don't think we would want to expose this on the API.

A much cheaper option would be to calculate this as a field in the `InputSelect<TValue>` type.

```
public class InputSelect<TValue> : InputBase<TValue>
{
    private readonly Type _nullableUnderlyingType = Nullable.GetUnderlyingType(typeof(TValue));
    ..
```

Reply:
Indeed ! I reverted that and used your solution.